File: ruby/lib/graphhopper/models/stop.rb
module GraphHopper
  # 
  class Stop < BaseObject
    attr_accessor :address, :duration, :time_windows
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # 
        :'address' => :'address',
        
        # duration of stop, i.e. time in ms the corresponding activity takes
        :'duration' => :'duration',
        
        # array of time windows. currently, only a single time window is allowed
        :'time_windows' => :'time_windows'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'address' => :'Address',
        :'duration' => :'Integer',
        :'time_windows' => :'Array<TimeWindow>'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'address']
        self.address = attributes[:'address']
      end
      
      if attributes[:'duration']
        self.duration = attributes[:'duration']
      end
      
      if attributes[:'time_windows']
        if (value = attributes[:'time_windows']).is_a?(Array)
          self.time_windows = value
        end
      end
      
    end

  end
end

File: dart/lib/model/stop.dart
part of swagger.api;


@Entity()
class Stop {
  
  @Property(name: 'address')
  Address address = null;
  
/* duration of stop, i.e. time in ms the corresponding activity takes */
  @Property(name: 'duration')
  int duration = null;
  
/* array of time windows. currently, only a single time window is allowed */
  @Property(name: 'time_windows')
  List<TimeWindow> timeWindows = [];
  
  Stop();

  @override
  String toString()  {
    return 'Stop[address=$address, duration=$duration, timeWindows=$timeWindows, ]';
  }

}

